docs/using-the-jdbc-driver/SessionState.md

```diff
@@ -0,0 +1,42 @@
+# Session States
+
+## What is session state
+
+Every connection is associated with a connection session on the server and related session settings like autoCommit flag or transaction isolation level. The following session settings are tracked by AWS JDBC Driver and forms a session state:
+- autoCommit (`setAutoCommit`, `getAutoCommit`)
+- readOnly (`isReadOnly`, `setReadOnly`)
+- transaction isolation level (`setTransactionIsolation`, `getTransactionIsolation`)
+- holdability (`setHoldability`, `getHoldability`)
+- network timeout (`setNetworkTimeout`, `getNetworkTimeout`)
+- catalog (`setCatalog`, `getCatalog`)
+- schema (`setSchema`, `getSchema`)
+- types mapping (`setTypeMap`, `getTypeMap`)
+
+Since AWS JDBC Driver can transparently switch physical connection to a server (for instance, during a cluster failover), it's important to re-apply a current session state to a new connection during such switch.   
```

Comment:
What does "transparently switch" mean?
```suggestion
Since the AWS JDBC Driver can transparently switch physical connection to a server (for instance, during a cluster failover), it's important to re-apply a current session state to a new connection during such switch.   
```